### PR TITLE
[SPARK-26929][SQL]fix table owner use user instead of principal when create table through spark-sql or beeline

### PR DESCRIPTION
**Pull request by hddong**
…create table through spark-sql or beeline

## What changes were proposed in this pull request?

fix table owner use user instead of principal when create table through spark-sql
`private val userName = conf.getUser`  will get ugi's userName which is principal info, and i copy the source code into HiveClientImpl, and use `ugi.getShortUserName()` instead of ` ugi.getUserName()`. The owner display correctly.

## How was this patch tested?

1. create a table  in kerberos cluster
2. use "desc formatted tbName" check owner

Please review http://spark.apache.org/contributing.html before opening a pull request.

### CODE REVIEW

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala [hunk 1]
```diff
@@ -186,7 +187,12 @@ private[hive] class HiveClientImpl(
   /** Returns the configuration for the current session. */
   def conf: HiveConf = state.getConf
 
-  private val userName = conf.getUser
+  private val userName: String = try {
+    val ugi = HiveUtils.getUGI
+    ugi.getShortUserName
+  } catch {
+    case e: LoginException => throw new IOException(e)
+  }
```
Comment (HyukjinKwon) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala]: Given this is a fix for forth and back (https://github.com/apache/spark/pull/20034 and https://github.com/apache/spark/pull/17363), can you test if the metastore is MySQL as well?

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala [hunk 2]
```diff
@@ -17,27 +17,27 @@
...
 import org.apache.hadoop.hive.metastore.{TableType => HiveTableType}
-import org.apache.hadoop.hive.metastore.api.{Database => HiveDatabase, FieldSchema, Order}
+import org.apache.hadoop.hive.metastore.api.{FieldSchema, Order, Database => HiveDatabase}
 import org.apache.hadoop.hive.metastore.api.{SerDeInfo, StorageDescriptor}
 import org.apache.hadoop.hive.ql.Driver
 import org.apache.hadoop.hive.ql.metadata.{Hive, Partition => HivePartition, Table => HiveTable}
 import org.apache.hadoop.hive.ql.parse.BaseSemanticAnalyzer.HIVE_COLUMN_ORDER_ASC
 import org.apache.hadoop.hive.ql.processors._
 import org.apache.hadoop.hive.ql.session.SessionState
-
```
Comment (HyukjinKwon) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala]: Please get rid of all unrelated changes.

**Comment by HyukjinKwon** [comment]
Can you fix the PR title to be complete (not truncated)? Also, please what code is referred and what is short name in PR description. This is a fix for a reverted change and usually it needs some details to avoid another forth and back.

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala [hunk 3]
```diff
@@ -17,27 +17,27 @@
...
 import scala.collection.mutable.ArrayBuffer
-
 import org.apache.hadoop.conf.Configuration
 import org.apache.hadoop.fs.Path
 import org.apache.hadoop.hive.common.StatsSetupConst
 import org.apache.hadoop.hive.conf.HiveConf
 import org.apache.hadoop.hive.conf.HiveConf.ConfVars
 import org.apache.hadoop.hive.metastore.{TableType => HiveTableType}
-import org.apache.hadoop.hive.metastore.api.{Database => HiveDatabase, FieldSchema, Order}
+import org.apache.hadoop.hive.metastore.api.{FieldSchema, Order, Database => HiveDatabase}
```
[the remark this answers is not in the document]
Comment (hddong) [reply sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala]: idea change this itself , not neccessary

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala (hunk 1 above)
Comment (hddong) [reply sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala]: > Given this is a fix for forth and back (#20034 and #17363), can you test if the metastore is MySQL as well?

it works correctly.

Comment (hddong) [reply sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala]: And i think it's not a fix for forth and back, actually, it is a copy of #20034  use ugi.shortUserName instead of userName

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala [hunk 4]
```diff
@@ -186,7 +187,12 @@ private[hive] class HiveClientImpl(
   /** Returns the configuration for the current session. */
   def conf: HiveConf = state.getConf
 
-  private val userName = conf.getUser
```
Comment (hddong) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala]: this source code of conf.getUser is:
`public String getUser() throws IOException {`
`        try {`
`             UserGroupInformation ugi = Utils.getUGI();` 
`             return ugi.getUserName();` 
`         } catch (LoginException var2) {` 
`             throw new IOException(var2);` 
`         }` 
`     }` 
this chang just use ugi.getShortUserName instead of ugi.getUserName()

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala (hunk 2 above)
Comment (hddong) [reply sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala]: > Please get rid of all unrelated changes.

All unrelated changes have been deleted.

**Comment by hddong** [comment]
@HyukjinKwon @felixcheung ,  could you help to review this again?

**Comment by HyukjinKwon** [comment]
@hddong, can you switch the target branch to master, not (branch-2.3)?

**Comment by hddong** [comment]
> @hddong, can you switch the target branch to master, not (branch-2.3)?

@HyukjinKwon ,  I changed it.

**Comment by HyukjinKwon** [comment]
Looks the base branch changed wrongly. Please feel free to create another PR against master branch if it's difficult to switch the branch.

**Comment by hddong** [comment]
A new PR was create at https://github.com/apache/spark/pull/23952/files.

**Comment by hddong** [comment]
@HyukjinKwon , could you help to review this again at new PR  https://github.com/apache/spark/pull/23952
